windbow27/Kornblume
# Feat: Introduce TypeScript, ESLint, and Vitest

## Description:

**Most of the changes are just formatter tweaks! No need to panic!**

This Pull Request introduces **TypeScript**, **ESLint**, and **Vitest** into the project, bringing several improvements and enhancements to the codebase, making it easier for us to refactor in the next steps.

- TypeScript: Enhances JavaScript with strong typing, which leads to better error checking, improved code readability, and easier maintainability. This is especially beneficial for large-scale or complex projects.

- ESLint: Ensures code quality and consistency by catching errors and enforcing best practices in JavaScript.

- Vitest: Offers fast and reliable testing for Vue.js applications, improving the overall testing process.

## Changes:

1. Restrict the range of versions for dependencies to ensure compatibility.
2. Introduce **TypeScript**:
   - Added a `tsconfig.json` file.
   - Adjusted the build script to incorporate TypeScript.
   - Converted the following Vue files to using TypeScript:
       1. App.vue
       2. Navbar.vue
       3. HomeView.vue
       4. PlannerView.vue
       5. ProfileView.vue
6. Introduce **ESLint**:
   - Added a `.eslintrc.cjs` configuration file.
   - Ran `eslint --fix` to automatically fix lint errors.
7. Introduce **Vitest** for efficient testing (no tests have been written yet)
8. Update the README to provide more detailed instructions on Recommended IDE Setup.

## How to Test
1. Pull and run `npm i` to install dependencies.
3. Follow the instructions in the README to set up the IDE:
     - [VSCode](https://code.visualstudio.com/)
     - [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) (and disable Vetur)
     - [TypeScript Vue Plugin (Volar)](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin)
     - [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
       **Strongly recommend setting up ESLint to auto-fix issues each time a file is saved, please check details [here](https://www.digitalocean.com/community/tutorials/workflow-auto-eslinting#step-4-adding-code-actions-on-save)**.
   
4. Run the script to check for any issues in development, build, or deployment.
